common/djangoapps/util/milestones_helpers.py

```diff
@@ -0,0 +1,151 @@
+            add_prerequisite_course(course_key, prerequisite_course_key)
+
+
+def get_pre_requisite_courses_not_completed(user, enrolled_courses):  # pylint: disable=invalid-name
+    """
+    It would make dict of prerequisite courses not completed by user among courses
+    user has enrolled in. It calls the fulfilment api of milestones app and
+    iterates over all fulfillments of user not achieved to make dict of
+    prerequisites yet to be completed.
+    """
+    pre_requisite_courses = {}
+    if settings.FEATURES.get('ENABLE_PREREQUISITE_COURSES'):
+        for course, __ in enrolled_courses:
+            required_courses = []
+            # if course has pre-requisites then fetch fulfilment path
+            if course.pre_requisite_courses:
+                fulfilment_paths = get_course_milestones_fulfillment_paths(course.id, {'id': user.id})
+                for milestone_key, milestone_value in fulfilment_paths.items():  # pylint: disable=unused-variable
+                    for key, value in milestone_value.items():
+                        if key == 'courses' and value:
+                            for required_course in value:
+                                required_course_key = CourseKey.from_string(required_course['course_id'])
+                                required_course_descriptor = modulestore().get_course(required_course_key)
+                                required_courses.append({
+                                    'key': required_course_key,
+                                    'display': ' '.join([
+                                        required_course_descriptor.display_org_with_default,
+                                        required_course_descriptor.display_number_with_default
+                                    ])
+                                })
+
+                # if there are required courses add to dict
+            if required_courses:
+                pre_requisite_courses[course.id] = {'courses': required_courses}
+    return pre_requisite_courses
+
+
+def get_prerequisite_courses_display(course_descriptor):  # pylint: disable=invalid-name
+    """
+    It would retrieve pre-requisite courses, make display strings
+    and return them as list
+    """
+    pre_requisite_courses = []
+    if settings.FEATURES.get('ENABLE_PREREQUISITE_COURSES', False) and course_descriptor.pre_requisite_courses:
+        for course_id in course_descriptor.pre_requisite_courses:
+            course_key = CourseKey.from_string(course_id)
+            required_course_descriptor = modulestore().get_course(course_key)
+            pre_requisite_courses.append(' '.join([
+                required_course_descriptor.display_org_with_default,
+                required_course_descriptor.display_number_with_default
+            ]))
+    return pre_requisite_courses
+
+
+def fulfill_course_milestone(course_key, user):
+    """
+    It would save course milestone collected by user.
```

Comment:
I don't understand what this method is doing. Is that what you call when a user completes a course? If so, something like this would be clearer:

Marks the course specified by the given course_key as complete for the given user. If any other courses require this course as a prerequisite, their milestones will be appropriately updated.